sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -293,8 +293,16 @@ trait BinaryNode extends LogicalPlan with BinaryLike[LogicalPlan]
 
 trait OrderPreservingUnaryNode extends UnaryNode
   with AliasAwareQueryOutputOrdering[LogicalPlan] {
-  override protected def outputExpressions: Seq[NamedExpression] = child.output
-  override protected def orderingExpressions: Seq[SortOrder] = child.outputOrdering
+
+  override def outputExpressions: Seq[NamedExpression] = child match {
```

Comment:
Why do we need these changes? AFAIK `OrderPreservingUnaryNode` means that the node doesn't change ordering. What would be the meaning of it after this cahnge?